### PR TITLE
get all list registrations for an account

### PR DESCRIPTION
**Pull request by Prometheo**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -148,6 +148,11 @@ enum PotApplicationStatus {
 }
 ```
 
+#### ✅ Get registrations to lists by account: `GET /accounts/{ACCOUNT_ID}/registrations` (paginated)
+
+Can specify status to filter by using `status` query param if desired, e.g. `status=Approved`
+Can also specify project category to filter by using `category` query param if desired, e.g. `category=Education`
```
Comment (lachlanglen) [inline README.md:154]: what is this? why would I want to filter by category when i'm getting registrations for a single project?

File: accounts/api.py [hunk 2]
```diff
@@ -391,3 +393,64 @@ def get(self, request: Request, *args, **kwargs):
...
+            if status_param not in ListRegistrationStatus.values:
+                return Response(
+                    {"message": f"Invalid status value: {status_param}"}, status=400
+                )
+            registrations = registrations.filter(status=status_param)
+        if category_param:
+            category_regex_pattern = rf'\[.*?"{category_param}".*?\]'
+            registrations = registrations.filter(
+                registrant__near_social_profile_data__plCategories__iregex=category_regex_pattern
+            )
```
Comment (lachlanglen) [inline accounts/api.py:453]: let's remove this, i don't see a use case here

File: api/urls.py [hunk 3]
```diff
@@ -76,6 +77,11 @@
         AccountPayoutsReceivedAPI.as_view(),
         name="accounts_api_by_id_payouts_received",
     ),
+    path(
+        "v1/accounts/<str:account_id>/registrations",
```
Comment (lachlanglen) [inline api/urls.py:81]: let's change to `"v1/accounts/<str:account_id>/list-registrations"` just to be extra clear, as we may have other kinds of registrations in the future

**Review by lachlanglen — CHANGES_REQUESTED** [review]
Looks pretty good, I think should just remove the `category` stuff (unless I'm crazy and not seeing something) as I don't see a use case for that. Also slight update to the API endpoint.

File: README.md (hunk 1 above)
Comment (Prometheo) [reply README.md:154]: i thought projects could apply to more than one list?

**Review by lachlanglen — APPROVED** [review]

**Pull request merged**